packages/keyring-controller/src/KeyringController.ts

```diff
@@ -635,9 +635,9 @@ export class KeyringController extends BaseController<
       await keyring.addAccounts(1);
       await this.#updateVault();
 
-      const addedAccountAddress = (await this.#getAccountsFromKeyrings()).find(
-        (selectedAddress) => !oldAccounts.includes(selectedAddress),
-      );
+      const addedAccountAddress = (await keyring.getAccounts())
+        .map(normalize)
+        .find((selectedAddress) => !oldAccounts.includes(selectedAddress));
```

Comment:
Since `keyring.addAccounts(1)` should return the created account, perhaps we can write this simply like:

```typescript
  const [addedAccountAddress] = await keyring.addAccounts(1);
  
  assertIsStrictHexString(addedAccountAddress);

  return addedAccountAddress;
```